MonoGame.Framework/Storage/StorageContainer.cs

```diff
@@ -0,0 +1,440 @@
+// MonoGame - Copyright (C) MonoGame Foundation, Inc
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using Microsoft.Xna.Framework;
+using System;
+using System.IO;
+
+#if WINDOWS_STOREAPP
+using Windows.Storage;
+using System.Linq;
+using Windows.Storage.Search;
+
+namespace System.IO
+{
+    public enum FileMode
+    {
+        CreateNew = 1,
+        Create = 2,
+        Open = 3,
+        OpenOrCreate = 4,
+        Truncate = 5,
+        Append = 6,
+    }
+
+    public enum FileAccess
+    {
+        Read = 1,
+        Write = 2,
+        ReadWrite = 3,
+    }
+
+    public enum FileShare
+    {
+        None = 0,
+        Read = 1,
+        Write = 2,
+        ReadWrite = 3,
+        Delete = 4,
+        Inheritable = 0x10,
+    }
+}
+#endif
+
+namespace Microsoft.Xna.Framework.Storage
+{
+    //	Implementation on Windows
+    //	
+    //	User storage is in the My Documents folder of the user who is currently logged in, in the SavedGames folder. 
+    //	A subfolder is created for each game according to the titleName passed to the BeginOpenContainer method. 
+    //	When no PlayerIndex is specified, content is saved in the AllPlayers folder. When a PlayerIndex is specified, 
+    //	the content is saved in the Player1, Player2, Player3, or Player4 folder, depending on which PlayerIndex 
+    //	was passed to BeginShowSelector.
+
+    /// <summary>
+    /// Contains a logical collection of files used for user-data storage.
+    /// </summary>			
+    /// <remarks>MSDN documentation contains related conceptual article: http://msdn.microsoft.com/en-us/library/bb200105.aspx#ID4EDB</remarks>
+    public class StorageContainer : IDisposable
+    {
+        internal readonly string _storagePath;
+        private readonly StorageDevice _device;
+        private readonly string _name;
+        private readonly PlayerIndex? _playerIndex;
```

Comment:
Do we want to commit the old implementation first?

Or should we gut this and/or start the new implementation?

Reply:
There just for the API, internal will 100% change.